File: backend/app/models/ingredient.rb
class Ingredient < ApplicationRecord
  has_many :meal_ingredients, dependent: :destroy
  has_many :meals, through: :meal_ingredients

  NUTRIENT_COLUMNS = %i[
    protein_per_100g
    fat_per_100g
    carbohydrate_per_100g
    sodium_per_100g
    calories_per_100g
  ].freeze

  validates :name, presence: true, uniqueness: true, length: { maximum: 100 }
  validates(*NUTRIENT_COLUMNS, numericality: { greater_than_or_equal_to: 0, allow_nil: true })
end

File: frontend/src/types/ingredient.ts
export type Ingredient = {
  id: number
  name: string
  protein_per_100g: string
  fat_per_100g: string
  carbohydrate_per_100g: string
  sodium_per_100g: string
  calories_per_100g: string
  created_at: string
  updated_at: string
}

export type IngredientCreateInput = Omit<Ingredient, 'id' | 'created_at' | 'updated_at'>
